### PR TITLE
Add renamehome command

### PR DESCRIPTION
**Pull request by kennytv**
### Information

This PR resolves https://github.com/EssentialsX/Essentials/discussions/4928. 

### Details

**Proposed feature:**  

Rename your homes

**Environments tested:**    

OS: Temple OS

Java version: 6

- [x] Most recent Paper version (1.19.84, git-Paper-191)
- [x] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**  

wooyeah

### CODE REVIEW

File: Essentials/src/main/resources/messages.properties [hunk 1]
```diff
@@ -1010,6 +1011,12 @@ removeCommandUsage1Description=Removes all of the given mob type in the current
 removeCommandUsage2=/<command> <mob type> <radius> [world]
 removeCommandUsage2Description=Removes the given mob type within the given radius in the current world or another one if specified
 removed=\u00a76Removed\u00a7c {0} \u00a76entities.
+renamehomeCommandDescription=Rename one of your homes.
+renamehomeCommandUsage=/<command> <[player:]name> <new name>
+renamehomeCommandUsage1=/<command> <name> <new name>
+renamehomeCommandUsage1Description=Renames your home with the given name
+renamehomeCommandUsage2=/<command> <player>:<name> <new name>
+renamehomeCommandUsage2Description=Renames the specified player's home with the given name
```
Comment (NoahvdAa) [inline Essentials/src/main/resources/messages.properties]:
```suggestion
renamehomeCommandUsage1Description=Renames your home with to given name
renamehomeCommandUsage2=/<command> <player>:<name> <new name>
renamehomeCommandUsage2Description=Renames the specified player's home to the given name
```
EDIT: nvm, misread that, with refers to the existing home, don't think I can resolve this myself tho...

Comment (kennytv) [reply Essentials/src/main/resources/messages.properties]: with to ฅʕ•̫͡•ʔฅ

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+public class Commandrenamehome extends EssentialsCommand {
+    public Commandrenamehome() {
+        super("renamehome");
+    }
+
+    @Override
+    public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
+        User usersHome = user;
+        final String name;
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java]: Nitpick; change to `oldName` for clarity?

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                throw new PlayerNotFoundException();
+            }
+
+            name = args[1].toLowerCase(Locale.ENGLISH);
+            newName = args[2].toLowerCase(Locale.ENGLISH);
+        } else {
+            throw new NotEnoughArgumentsException();
+        }
+
+        if ("bed".equals(newName) || NumberUtil.isInt(newName) || "bed".equals(name) || NumberUtil.isInt(name)) {
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java]: Is the case-sensitivity here consistent with other home commands? (ie check whether `/home bEd` takes you to your bed or just `/home bed`)

File: Essentials/src/main/resources/messages.properties [hunk 4]
```diff
@@ -1010,6 +1011,12 @@ removeCommandUsage1Description=Removes all of the given mob type in the current
 removeCommandUsage2=/<command> <mob type> <radius> [world]
 removeCommandUsage2Description=Removes the given mob type within the given radius in the current world or another one if specified
 removed=\u00a76Removed\u00a7c {0} \u00a76entities.
+renamehomeCommandDescription=Rename one of your homes.
```
Comment (JRoy) [inline Essentials/src/main/resources/messages.properties]:
```suggestion
renamehomeCommandDescription=Rename a home.
```

**Review by JRoy — CHANGES_REQUESTED** [review]

File: Essentials/src/main/resources/messages.properties (hunk 4 above)
Comment (mdcfe) [reply Essentials/src/main/resources/messages.properties]: Or `Renames`, whichever is more consistent with other lang strings

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java (hunk 3 above)
Comment (kennytv) [reply Essentials/src/main/java/com/earth2me/essentials/commands/Commandrenamehome.java]: Yes, case on bed is ignored on home and delhome

**Review by mdcfe — APPROVED** [review]

**Review by JRoy — APPROVED** [review]

**Comment by seema84** [comment]
is there a permission for this?

**Pull request merged**